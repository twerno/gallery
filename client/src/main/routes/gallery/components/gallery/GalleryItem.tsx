import * as React from 'react';
import { IResponsivePictureProps } from './GalleryResponsivePicture';
import { LazyResponsiveGalleryPicture } from './LazyResponsiveGalleryPicture';
import GalleryStyles from './styles/GalleryStyles';

export interface IGalleryItemProps {
    imageIdx: number;
    disabled?: boolean;
    responsivePicture: IResponsivePictureProps;
}

export const GalleryItem = (props: IGalleryItemProps) => {

    // const dispatch = useDispatch();
    // const clickHandler =
    //     () => dispatch(galleryItemSlice.actions.showFullImg(props.imageIdx));

    const ref = React.createRef<HTMLDivElement>();
    return (
        <GalleryStyles.GalleryItem
            ref={ref}
            disabled={props.disabled}
        // onClick={clickHandler}
        >
            <LazyResponsiveGalleryPicture
                containerRef={ref}
                pictureData={props.responsivePicture}
            />
        </GalleryStyles.GalleryItem>
    );
};
