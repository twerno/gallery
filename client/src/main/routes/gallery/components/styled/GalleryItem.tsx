import { breakpoints } from 'main/globalStyles';
import styled, { css } from 'styled-components';

export interface IGalleryItemProps {
    disable?: boolean;
}


export const GalleryItem = styled.div<IGalleryItemProps>`
    margin: 1px;
    flex-grow: 1;
    overflow: hidden;
    display: flex;
    height: 200px;

    @media only screen and (max-width: ${breakpoints.md}) {

        & {
            height: 100px;
        }
    }

    pointer-events: ${props => props.disable ? 'none' : 'initial'};

    picture {
        min-width: 100%;
    }

    img {
        object-fit: cover;
        min-width: 100%;
        height: 100%;
    }
`;

/**
 * 
 */
export const FullScreenGalleryItem = styled.div<IGalleryItemProps>`
    position: absolute;
    display: flex;
    justify-content: center;
    align-items: center;

    width: 100%;
    height: 100%;
    padding: 2rem;

    picture {
        height: 100%;
    }

    img {
        height: 100%;
        max-width: 100%;
        object-fit: contain;
    }
`;

/**
 * 
 */
interface IHideablePictureProps {
    readonly isHidden: boolean;
}

/**
 * 
 */
export const HideablePicture = styled.picture<IHideablePictureProps>`
    opacity: 1;
    position: relative;
    transition: opacity 0.2s ease-in-out;

    ${
    props => props.isHidden && css`
            position: absolute;
            opacity: 0;
    `}
`;
