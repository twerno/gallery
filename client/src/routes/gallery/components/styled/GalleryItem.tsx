import styled from 'styled-components';

export default styled.div`
    margin: 1px;
    flex-grow: 1;
    overflow: hidden;
    display: flex;
    height: 200px;

    @media only screen and (max-width: 600px) {

        & {
            height: 100px;
        }
    }
`;
