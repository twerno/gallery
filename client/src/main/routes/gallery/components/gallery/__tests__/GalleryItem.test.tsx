import '@testing-library/jest-dom/extend-expect';
import 'jest-styled-components';
import * as React from 'react';
import { componentBasicTests } from 'test/componentBasicTests';
import { GalleryItem, IGalleryItemProps } from '../GalleryItem';

describe('GalleryItem', () => {

    const props: IGalleryItemProps = {
        disabled: true,
        imageIdx: 0,
        responsivePicture: {
            alt: 'alt',
            defaultImgSrc: 'img.jpg',
            responsiveSrcList: [{ src: 'img-big.jpg', media: '(min-width: 600px)' }]
        },
    };

    componentBasicTests(
        "GalleryItem - disabled: false",
        <GalleryItem {...props} disabled={false} />
    );

    componentBasicTests(
        "GalleryItem - disabled: true",
        <GalleryItem {...props} disabled={true} />
    );

    componentBasicTests(
        "GalleryItem - imageIdx: 1",
        <GalleryItem {...props} imageIdx={1} />
    );
});

