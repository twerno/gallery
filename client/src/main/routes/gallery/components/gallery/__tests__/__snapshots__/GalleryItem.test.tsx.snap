// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GalleryItem [GalleryItem - disabled: false] Component basic tests matches the snapshot 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 500px;
  width: 100%;
  height: 100%;
}

.c0 {
  margin: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  pointer-events: initial;
}

.c2 {
  background-color: rgba(0,0,0,0.05);
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c2::before {
  border: .25rem solid rgba(0,0,0,0.3);
  -webkit-animation: 4s infinite spin linear;
  animation: 4s infinite spin linear;
}

.c2::before,
.c2::after {
  content: '';
  position: absolute;
  width: 2rem;
  height: 2rem;
  border-right-color: transparent;
  border-left-color: transparent;
  border-radius: 50%;
}

@media only screen and (max-width:600px) {
  .c1 {
    min-width: 200px;
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    height: 100px;
  }
}

<div
  className="c0"
  disabled={false}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
  </div>
</div>
`;

exports[`GalleryItem [GalleryItem - disabled: true] Component basic tests matches the snapshot 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 500px;
  width: 100%;
  height: 100%;
}

.c0 {
  margin: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  pointer-events: none;
}

.c2 {
  background-color: rgba(0,0,0,0.05);
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c2::before {
  border: .25rem solid rgba(0,0,0,0.3);
  -webkit-animation: 4s infinite spin linear;
  animation: 4s infinite spin linear;
}

.c2::before,
.c2::after {
  content: '';
  position: absolute;
  width: 2rem;
  height: 2rem;
  border-right-color: transparent;
  border-left-color: transparent;
  border-radius: 50%;
}

@media only screen and (max-width:600px) {
  .c1 {
    min-width: 200px;
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    height: 100px;
  }
}

<div
  className="c0"
  disabled={true}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
  </div>
</div>
`;

exports[`GalleryItem [GalleryItem - imageIdx: 1] Component basic tests matches the snapshot 1`] = `
.c1 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-width: 500px;
  width: 100%;
  height: 100%;
}

.c0 {
  margin: 1px;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 200px;
  pointer-events: none;
}

.c2 {
  background-color: rgba(0,0,0,0.05);
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.c2::before {
  border: .25rem solid rgba(0,0,0,0.3);
  -webkit-animation: 4s infinite spin linear;
  animation: 4s infinite spin linear;
}

.c2::before,
.c2::after {
  content: '';
  position: absolute;
  width: 2rem;
  height: 2rem;
  border-right-color: transparent;
  border-left-color: transparent;
  border-radius: 50%;
}

@media only screen and (max-width:600px) {
  .c1 {
    min-width: 200px;
  }
}

@media only screen and (max-width:600px) {
  .c0 {
    height: 100px;
  }
}

<div
  className="c0"
  disabled={true}
>
  <div
    className="c1"
  >
    <div
      className="c2"
    />
  </div>
</div>
`;
